autogen/src/dr.rs

```diff
@@ -308,6 +332,221 @@ pub fn gen_dr_operand_kinds(grammar: &[structs::OperandKind]) -> TokenStream {
                 }
             }
         });
+
+        let operand_metadata = kind_to_enum
+            .iter()
+            .filter_map(|(kind, category, enumerators)| {
+                if enumerators.is_empty() {
+                    return None;
+                }
+
+                let mut capability_clauses = BTreeMap::new();
+                let mut extension_clauses = BTreeMap::new();
+                let mut operand_clauses = BTreeMap::new();
+
+                let kind = as_ident(kind);
+                let mut seen_discriminator = BTreeMap::new();
+
+                for e in enumerators {
+                    if seen_discriminator.get(&e.value).is_none() {
+                        let name = match category {
+                            structs::Category::BitEnum => {
+                                use heck::ShoutySnakeCase;
+
+                                as_ident(&e.symbol.to_shouty_snake_case().replace("NA_N", "NAN"))
+                            }
+                            structs::Category::ValueEnum => {
+                                let name_str = if kind == "Dim" {
+                                    let mut name = "Dim".to_string();
+                                    name.push_str(&e.symbol);
+                                    name
```

Comment:
I wonder how much faster this is in than `format("Dim{}", &e.symbol)`. It's been a while since I last went through Rusts formatting internals, but that should expand to `.push_str` as well and `Display::fmt` on a string reference like `&e.symbol` should be a cheap reference.